proposals/nnnn-Use-Taskmaster-To-Handle-Queuing-Operations-In-Managers.md

```diff
@@ -0,0 +1,80 @@
+
+## Proposed solution
+
+The proposal suggests using [`TaskMaster`](https://github.com/livio/Taskmaster) in SDL Java Suite to create serial queues of operations that need to be executed in order. `ChoiceSetManager`, currently, uses `ThreadPoolExecutor` to queue operations and run them in order. For example, it uses a queue to submit `CheckChoiceVROptionalOperation` then `PresentChoiceSetOperation` to make sure that manager doesn't start presenting choices before knowing whether VR is optional or not on the target head unit. This queuing approach is much better than manually tracking if there is something queued (as the library currently still does  that in `SoftButtonManager` for example). However, the `ThreadPoolExecutor` solution falls short when we need to extend that solution to more managers because each manager and potentially submanagers will be forced to create new threads.
+
+[`TaskMaster`](https://github.com/livio/Taskmaster) is queue oriented thread management solution developed by the SDL project maintainer to solve aforementioned issue. It is designed in a way that allows each manager to have its own queue of operations but still run all queues on a fixed number of threads. This gets rid of the thread-per-queue restriction that the `ThreadPoolExecutor` solution suffers. SDL iOS already uses `NSOperationQueue` to manage queuing operations in the managers which is a similar solution but it handles threads at the OS level.
+
+The proposal suggests adding an API called `getTaskmaster()` to the `Isdl` interface which allows all managers to access the same `Taskmaster` instance:
+
+```
+public interface ISdl {
+    ...
+
+    Taskmaster getTaskmaster();
+}
+```    
+
+Each manager that uses queues can create its own queue using the `Taskmaster` instance and add operations to the queue to be executed it order:
+
+```
+class ChoiceSetManager {
+    Queue operationQueue;
+
+    ChoiceSetManager (){
+        operationQueue = internalInterface.getTaskmaster().createQueue("operationQueue", 1, false);
```

Comment:
Do any "Taskmaster" references within the code need to use "TaskMaster" capitalization?